### PR TITLE
Notifications: Move to apps/

### PR DESCRIPTION
**Pull request by ockham**
Will come in handy for #32601.

#### Changes proposed in this Pull Request

Notifications: Move to apps/

They seem to be a perfect example of our `apps/` paradigm: 
- Only one (style) `import` of Calypso code AFAICS
- `webpack.config.js` of their own
- Built through `calypso-build`

#### Testing instructions

```
npm run distclean && npm ci
npx lerna run build 
```

Verify that notifications build. Furthermore, follow testing instructions found at PCYsg-elI-p2

```
npm start
```

Verify that Calypso still works. Specifically, smoke-test notifications.

### CODE REVIEW

File: client/layout/masterbar/notifications.jsx [hunk 1]
```diff
@@ -54,14 +54,14 @@ class MasterbarItemNotifications extends Component {
 
 		// focus on main window if we just closed the notes panel
 		if ( this.props.isNotificationsOpen && ! isOpen ) {
-			this.getNotificationLinkDomNode().blur();
+			this.notificationLink.current.blur();
```
Comment (sirreal) [inline client/layout/masterbar/notifications.jsx]: Does this need a check on `current` being truthy before calling `.blur()`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -274,7 +274,7 @@ jobs:
       - run:
           name: Build Notifications Panel
           command: |
-            NODE_ENV=production npm run build-notifications -- --output-path=$CIRCLE_ARTIFACTS/notifications-panel
+            NODE_ENV=production npx lerna run build --scope='@automattic/notifications' -- -- --output-path=$CIRCLE_ARTIFACTS/notifications-p
```
Comment (sirreal) [inline .circleci/config.yml]: This was mistakenly truncated somehow which screws up the artifacts:

```suggestion
            NODE_ENV=production npx lerna run build --scope='@automattic/notifications' -- -- --output-path=$CIRCLE_ARTIFACTS/notifications-panel
```

Aside: anything built with `calypso-build` that's using `output-dir` to the artifact directory does not need an entry in `setup-results-and-artifacts` like `        "$CIRCLE_ARTIFACTS/notifications-panel"`, we could remove several lines as the directory is created on demand.

**Review by sirreal — DISMISSED** [review]
Blocker and a question.

I wonder if the change from string ref to `createRef` is that simple or we should add more safety…

File: client/layout/masterbar/notifications.jsx (hunk 1 above)
Comment (jsnajdr) [reply client/layout/masterbar/notifications.jsx]: `this.notificationLink.current` is guaranteed to be a non-null DOM node. The ref is attached in the render method:
```jsx
<div ref={ this.notificationLink } >
```
The `div` is always rendered, there is no conditional rendering that would sometimes render it and sometimes not. And at the time `componentWillReceiveProps` is called, the mounting and ref calling is guaranteed to be done.

**Review by sirreal — APPROVED** [review]
This is working well and seems to be compatible with WPCOM-side scripts handling the build artifact.

I'm able to observe updated colors (the panel could use an update):

Before:

before


After:

after

**Comment by kwight** [comment]
Sweet, looks like this works well 🎉 

I noticed three warnings during the build:
- `Entrypoint undefined = cache-buster.txt`
- `Entrypoint undefined = index.html`
- `Entrypoint undefined = rtl.html`

And just to be sure I'm testing correctly, I tested the deploy instructions with the build ID from this PR, not from the result of the `npx lerna run build` command (into `dist`) – correct?

I also diffed the output of the lerna build and the CircleCI build, and it looks like the CircleCI build is minified, while the other is not – I assume that's expected?

Thanks for doing this ✨

**Review by kwight — APPROVED** [review]

**Comment by ockham** [comment]
> Sweet, looks like this works well 
> 
> I noticed three warnings during the build:
> 
> * `Entrypoint undefined = cache-buster.txt`
> * `Entrypoint undefined = index.html`
> * `Entrypoint undefined = rtl.html`

I saw these too. I'll admit that I'm not entirely sure what to make of these, but my hunch was that they were simply listed as entrypoints that weren't assigned a name in the webpack config -- which to me didn't seem problematic (especially as the build seems to work fine).

> And just to be sure I'm testing correctly, I tested the deploy instructions with the build ID from this PR, not from the result of the `npx lerna run build` command (into `dist`) – correct?

Yep, that's correct! Sorry, I should've been clearer in my testing instructions :sweat_smile: 

> I also diffed the output of the lerna build and the CircleCI build, and it looks like the CircleCI build is minified, while the other is not – I assume that's expected?

Yep, the CircleCI build prepends `NODE_ENV=production` to the lerna command. (Doing so locally should repro minification.)

> Thanks for doing this :sparkles: 

My pleasure! Thanks for reviewing!

File: .circleci/config.yml [hunk 3]
```diff
@@ -274,7 +274,7 @@ jobs:
       - run:
           name: Build Notifications Panel
           command: |
-            NODE_ENV=production npx lerna run build --scope='@automattic/notifications' -- -- --output-path=$CIRCLE_ARTIFACTS/notifications-p
```
Comment (ockham) [inline .circleci/config.yml]: D'oh, thanks! No idea what happened there :sweat_smile:

**Comment by dmsnell** [comment]
> I saw these too. I'll admit that I'm not entirely sure what to make of these, but my hunch was that they were simply listed as entrypoints that weren't assigned a name in the webpack config -- which to me didn't seem problematic (especially as the build seems to work fine).

Did you confirm that these three files are being generated? They are the artifacts deployed to the iframed version of the app in the masterbar.

**Comment by kwight** [comment]
> Did you confirm that these three files are being generated?

They are, yes 👍

**Pull request merged**